File: xide-standard-plugins/xide-scene-editor-plugin/src/main/java/org/xast/xide/scene_editor_plugin/SceneEditorPlugin.java
package org.xast.xide.scene_editor_plugin;

import java.io.File;

import org.xast.xide.core.event.EventBus;
import org.xast.xide.core.plugin.file.FilePlugin;
import org.xast.xide.core.plugin.ui.CodePanelView;

public class SceneEditorPlugin implements FilePlugin {
    @Override
    public String[] fileExtensions() {
        return new String[] { "xsc" };
    }

    @Override
    public CodePanelView view(EventBus eventBus, File file) {
        return new SceneEditorView(eventBus, file);
    }
}

File: xide-standard-plugins/xide-scene-editor-plugin/src/main/kotlin/org/xast/xide/scene_editor_plugin/SceneEditorPlugin.kt
package org.xast.xide.scene_editor_plugin

import java.io.File

import org.xast.xide.core.event.EventBus
import org.xast.xide.core.plugin.file.FilePlugin
import org.xast.xide.core.plugin.ui.CodePanelView

class SceneEditorPlugin : FilePlugin {
    override fun fileExtensions(): Array<String> = arrayOf("xsc")

    override fun view(eventBus: EventBus, file: File): CodePanelView =
        SceneEditorView(eventBus, file)
}
